### DISCUSSION TITLE
wildcard for *.log files in generated directories

### DISCUSSION THREAD

**Discussion by tanguy-otiv** [syslog-ng/syslog-ng#4822] [General]
Hello everyone,

I'm having an issue with syslog-ng version 4.6.0-1 on Ubuntu 22.04. I'm trying to use the wildcard-file source to detect new log files in newly generated directories, but it doesn't seem to be working as expected. Here's the configuration I'm using:

source s_src {
    wildcard-file(
        base-dir("/home/user/ROS2/data/live_sensor_logs/")
        filename-pattern("*.log")
        flags(no-parse)
        recursive(yes)
        follow-freq(1)
    );
};
The wildcard-file source is not detecting new log files in newly generated directories. Any ideas on how to fix this?

Thanks in advance for your help!

**Comment by tanguy-otiv** [comment]
It seems to be a permission issue but I don't understand why, syslog-ng is run by user root:
Feb 8 11:51:20 otiv-two-1 syslog-ng[265779]: Can't resolve to absolute path; path='/home/user/ROS2/data/', error='Permission denied (13)'
Feb 8 11:51:20 otiv-two-1 syslog-ng[265779]: Can not open directory; base_dir='(null)', error='Error opening directory “(null)”: Permission denied'

**Comment by czanik** [comment]
You might have AppArmor or SELinux enabled, which limit syslog-ng to specific directories.

Also, I have some distant memories, that capabilities support can prevent you from reading other user's files. The --no-caps parameter takes care of that.

Or a combination of the two...

**Comment by tanguy-otiv** [comment]
Hello czanik, I stopped AppArmor but it did not change anything.
I've set the SYSLOGNG_OPTS="--no-caps" and that did the trick, thank you!
But now I'm having:
Number of allowed monitorod file is reached, rejecting read file; source='s_adas', filename='/home/ ... /init_gps.log', max_files='100'

**Comment by bazsi** [comment]
I literally just submitted a PR to improve that ugly error message.

 You need to increase the max-files() option. Which limits the number of log files.

The default limit is 100 files

**Reply by bazsi** [reply]
@tangoy-otiv did the limit increase resolve your issue?

**Reply by tanguy-otiv** [reply]
Hello @bazsi, as soon as I find the time, I will test and come back to you.
Already, thank you for your work.
